sxw_resource.c

```diff
@@ -237,29 +203,25 @@ static void _transp_contribution_by_group(RealF use_by_group[]) {
     LyrIndex l;
     RealD *transp, proportion_total_resources, average_proportion;
     RealF sumUsedByGroup = 0., sumTranspTotal = 0., TranspRemaining = 0.;
-    RealF transp_ratio, added_transp = 0, transp_ratio_sd;
+    RealF transp_ratio, transp_ratio_sd;
+    transp_window->added_transp = 0;
 
     // year 0 is a set up year. No need to calculate transpiration.
     // if there are multiple iterations the last year will run twice;
     // once for data and once for tear down. The second "last year" is
     // equivalent to year 0.
-    if(Globals.currYear == 0 || Globals.currYear == lastYear) {
-      transp_window.average = 0;
-      transp_window.ratio_average = 0;
-      transp_window.sum_of_sqrs = 0;
-      if(Globals.transp_window > MAX_WINDOW){
-                LogError(logfp, LOGNOTE, "sxw_resource: Transp_window specified in inputs is greater than maximum window.\nInput: %d\nMaximum: %d\nSetting window to %d.\n",Globals.transp_window, MAX_WINDOW,MAX_WINDOW);
-                Globals.transp_window = MAX_WINDOW;
-      }
```

Comment:
@kpalmqui the LogError has been moved to _allocate_memory in sxw.c(). It made more sense to place it in _allocate_memory because it is only called once, whereas _transp_contribution_by_group is called multiple times. transp_window only has to be checked and set once.